backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts

```diff
@@ -0,0 +1,34 @@
+import * as express from 'express';
+import { SettingService } from '../services/SettingService';
+import { SERVER_ERROR } from 'upgrade_types';
+import { AppRequest } from '../../types';
+import { Service } from 'typedi';
+import { ExperimentUserService } from '../services/ExperimentUserService';
+
+@Service()
+export class UserCheckMiddleware {
+  constructor(public settingService: SettingService, public experimentUserService: ExperimentUserService) {}
+
+  public async use(req: AppRequest, res: AppRequest, next: express.NextFunction): Promise<any> {
+    try {
+      const user_id = req.get('User-Id');
+      req.logger.child({ user_id });
+      req.logger.debug({ message: 'User Id is:', user_id });
+      
+      const experimentUserDoc = await this.experimentUserService.getUserDoc(user_id, req.logger);
+      if (req.url !== '/api/v6/init' && !experimentUserDoc) {
```

Comment:
does this need to be a hard-coded v6 in the endpoint for version check? This middleware should be version agnostic.

Reply:
Okay, got it. Will update this with any other reviews I receive today.